Update overview page

song50119 opened the pull request:
Resolves #334

song50119 committed:
Update overview page
- add wallet name to transactions on the overview page
- apply proper margins to the sync status info and icons
- line separator on recent transactions shouldn not span the entire width of the recent transactions card.
- add text button to show and hide sync details
- switch sync details from horizontal to vertical layout
- add a line separator below "wallet status" title
- increase the radius of the sync trigger button
- change sync button to icon button and change button text from "Connect" to "reconnect" when not connected

oshorefueled commented on ui/page.go:
```diff
@@ -30,7 +30,7 @@ type pageIcons struct {
 	arrowForwardIcon, transactionFingerPrintIcon, settingsIcon, securityIcon, helpIcon,
 	aboutIcon, debugIcon, verifyMessageIcon, locationPinIcon, alertGray, arrowDownIcon,
-	watchOnlyWalletIcon *widget.Image
+	watchOnlyWalletIcon, rescan *widget.Image
```
There seems to be an iconvg equivalent of this icon

song50119 committed:
Correct style overview page
- Change spacing of sysn status icon
- Delete rescan img icon and use cached icon from materialdesign instead
- Display wallet name on txn row if have multiple wallets
- Add recent txn row fixed height
- Change name recentTransactionsColumn to recentTransactionsSection, syncStatusColumn to syncStatusSection
- Add walletBadge to show wallet name on each row of txn

oshorefueled commented on ui/overview_page.go:
```diff
@@ -309,55 +271,185 @@ func (pg *overviewPage) centralize(gtx layout.Context, content layout.Widget) la
+															txnWidgets.wallet.Color = pg.theme.Color.Gray
+															return txnWidgets.wallet.Layout(gtx)
+														})
+													})
```
This transaction row can be broken down into smaller methods or functions to make it easier to visualize the layout of the UI with a quick glance at the code. This for instance can be kept in a method with a name "walletBadge".

oshorefueled commented on ui/overview_page.go:
```diff
@@ -309,55 +271,185 @@ func (pg *overviewPage) centralize(gtx layout.Context, content layout.Widget) la
 				layout.Rigid(func(gtx C) D {
-					return layout.Inset{Top: values.MarginPadding10, Left: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
-						return txn.statusIcon.Layout(gtx)
+					return layout.Inset{
+						Left:   values.MarginPadding40,
+						Top:    values.MarginPadding0,
+						Bottom: values.MarginPadding0,
+					}.Layout(gtx, func(gtx C) D {
+						pg.line.Width = gtx.Constraints.Max.X
+						return pg.line.Layout(gtx)
```
This is a separator

oshorefueled commented on ui/overview_page.go:
```diff
@@ -309,55 +271,185 @@ func (pg *overviewPage) centralize(gtx layout.Context, content layout.Widget) la
-					return layout.Inset{Top: values.MarginPadding10}.Layout(gtx, func(gtx C) D {
-						return txn.status.Layout(gtx)
+					heightWithResolution := pg.txnRowHeight * int(gtx.Metric.PxPerDp)
+					gtx.Constraints.Min.Y, gtx.Constraints.Max.Y = heightWithResolution, heightWithResolution
```
kindly add a comment to code like this that's not a common UI layout or modification

oshorefueled commented on ui/overview_page.go:
```diff
@@ -309,55 +271,185 @@ func (pg *overviewPage) centralize(gtx layout.Context, content layout.Widget) la
 }
 
 // syncStatusColumn lays out content for displaying sync status.
 func (pg *overviewPage) syncStatusColumn(gtx layout.Context) layout.Dimensions {
```
kindly rename this to "syncStatusSection". It's a section and not a column

oshorefueled commented on ui/overview_page.go:
```diff
@@ -206,53 +219,20 @@ func (pg *overviewPage) syncDetail(name, status, headersFetched, progress string
 // recentTransactionsColumn lays out the list of recent transactions.
 func (pg *overviewPage) recentTransactionsColumn(gtx layout.Context, c pageCommon) layout.Dimensions {
```
kindly rename this to "recentTransactionsSection"

oshorefueled commented on ui/overview_page.go:
```diff
@@ -366,28 +458,13 @@ func (pg *overviewPage) syncStatusColumn(gtx layout.Context) layout.Dimensions {
+	return uniform.Layout(gtx, func(gtx C) D {
 		return layout.Flex{Axis: layout.Vertical}.Layout(gtx,
 			layout.Rigid(func(gtx C) D {
 				return layout.Inset{Bottom: values.TextSize12}.Layout(gtx, func(gtx C) D {
```
TextSize dimension shouldn't be used for margin or padding. Kindly define an equivalent of this dimension as MarginPadding

oshorefueled commented on ui/overview_page.go:
```diff
@@ -486,15 +563,13 @@ func (pg *overviewPage) syncBoxTitleRow(gtx layout.Context, inset layout.Inset)
-							return layout.Inset{Top: values.MarginPadding5, Right: values.MarginPadding20}.Layout(gtx, func(gtx C) D {
-								return pg.walletStatusIcon.Layout(gtx, values.MarginPadding10)
+							return layout.Inset{Top: values.MarginPadding4, Right: values.MarginPadding4}.Layout(gtx, func(gtx C) D {
+								return pg.walletStatusIcon.Layout(gtx, values.TextSize14)
```
TextSize dimension shouldn't be used for margin or padding. Kindly define an equivalent of this dimension as MarginPadding

oshorefueled commented on ui/overview_page.go:
```diff
@@ -551,7 +624,23 @@ func (pg *overviewPage) syncStatusTextRow(gtx layout.Context, inset layout.Inset
+		})
+	}
+	return i.Layout(gtx, func(gtx C) D {
+		return syncStatusIcon.Layout(gtx, values.TextSize24)
```
TextSize dimension shouldn't be used for margin or padding. Kindly define an equivalent of this dimension as MarginPadding

oshorefueled commented on ui/overview_page.go:
```diff
@@ -748,10 +842,10 @@ func layoutBalance(gtx layout.Context, amount string, c pageCommon) layout.Dimen
 	return layout.Flex{Axis: layout.Horizontal, Alignment: layout.Baseline}.Layout(gtx,
 		layout.Rigid(func(gtx C) D {
-			return c.theme.H6(mainText).Layout(gtx)
+			return c.theme.Label(values.MarginPadding20, mainText).Layout(gtx)
```
Kindly define an equivalent of this dimension as TextSize. Text size should be defined as "sp" units rather than "dp"
